spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.amqp.rabbit.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.amqp.AmqpException;
+import org.springframework.amqp.core.AmqpManagementOperations;
+import org.springframework.amqp.core.Binding;
+import org.springframework.amqp.core.Binding.DestinationType;
+import org.springframework.amqp.core.DirectExchange;
+import org.springframework.amqp.core.Exchange;
+import org.springframework.amqp.core.FanoutExchange;
+import org.springframework.amqp.core.HeadersExchange;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.TopicExchange;
+
+import com.rabbitmq.http.client.Client;
+import com.rabbitmq.http.client.domain.BindingInfo;
+import com.rabbitmq.http.client.domain.ExchangeInfo;
+import com.rabbitmq.http.client.domain.QueueInfo;
+
+/**
+ * A convenience wrapper for the RabbitMQ {@link Client} providing convenient access to
+ * the REST methods using the familiar Spring AMQP domain objects for {@link Queue},
+ * {@link Exchange} and {@link Binding}. For more complete access, including access to
+ * properties not available in the Spring AMQP domain classes, use the {@link Client}
+ * directly.
+ *
+ * @author Gary Russell
+ * @since 1.5
+ *
+ */
+public class RabbitManagementTemplate implements AmqpManagementOperations {
+
+	private static final String DEFAULT_VHOST = "/";
+
+	private final Client rabbitClient;
+
+
+	/**
+	 * Construct a template using uri "localhost:15672/api/" and user guest/guest.
+	 */
+	public RabbitManagementTemplate() {
+		this("http://localhost:15672/api/", "guest", "guest");
+	}
+
+	/**
+	 * Construct a template using the supplied client.
+	 * @param rabbitClient the client.
+	 */
+	public RabbitManagementTemplate(Client rabbitClient) {
+		this.rabbitClient = rabbitClient;
+	}
+
+	/**
+	 * Construct a template using the supplied uri.
+	 * @param uri the uri.
+	 * @param username the user.
+	 * @param password the password.
+	 */
+	public RabbitManagementTemplate(String uri, String username, String password) {
+		try {
+			this.rabbitClient = new Client(uri, username, password);
+		}
+		catch (Exception e) {
+			throw new AmqpException(e);
```

Comment:
Let's discuss this part.
Really not sure that we should wrap any underlying exception to such a specific `AmqpException`.
What is the concern do rethrow as is? Just make it more friendly for end-users?